server/handlers/handlers.go

```diff
@@ -563,3 +563,38 @@ func (ah *ActivityHandler) isRemoteUser(userID string) bool {
 func IsDirectMessage(chatID string) bool {
 	return chatID != ""
 }
+
+func (ah *ActivityHandler) SyncChannelSince(teamID, channelID string, syncSince time.Time) error {
+	messages, err := ah.plugin.GetClientForApp().ListChannelMessages(teamID, channelID, syncSince)
+	if err != nil {
+		return err
+	}
+	for _, message := range messages {
```

Comment:
We can do explicit pagination, but I think the pagination is inside the client method, so I get the whole list of messages at this point and loop over it. Are you concern about the amount of memory usage?

Reply:
I think it is a valid concern.  Memory ballooning on the server can bring down the whole server.  Since this is unbounded, someone can specify a channel should be sync'd from years ago and get many thousands of messages.

Reply:
I going to change the approach and use something like a channel to return the values there.

Reply:
I finally decided to keep it simple: pass a callback function and rename the function to OnChannelMessageSince, which receives a callback that receives one call per message. It is not "all or nothing", but I don't think it is critical in this case because it is going to be idempotent anyway.

Reply:
Done @wiggin77 PTAL